services/friendbot/init_friendbot.go

```
@@ -13,24 +15,71 @@ func initFriendbot(
 	networkPassphrase string,
 	horizonURL string,
 	startingBalance string,
+	numMinions int,
 ) *internal.Bot {
-
-	if friendbotSecret == "" || networkPassphrase == "" || horizonURL == "" || startingBalance == "" {
+	if friendbotSecret == "" || networkPassphrase == "" || horizonURL == "" || startingBalance == "" || numMinions < 0 {
 		return nil
 	}
 
 	// ensure its a seed if its not blank
 	strkey.MustDecode(strkey.VersionByteSeed, friendbotSecret)
 
+	hclient := &horizon.Client{
+		URL:     horizonURL,
+		HTTP:    http.DefaultClient,
+		AppName: "friendbot",
+	}
+
+	minions := make([]internal.Minion, numMinions)
+	for i := 0; i < numMinions; i++ {
+		minionSecret, err := createNewAccount(friendbotSecret, networkPassphrase, hclient)
+		if err != nil {
+			return nil
+		}
+		minions = append(minions, internal.Minion{
+			Secret:       minionSecret,
+			DestAddrChan: make(chan string),
+			TxResultChan: make(chan internal.TxResult),
+		})
+	}
+
 	return &internal.Bot{
-		Secret: friendbotSecret,
-		Horizon: &horizon.Client{
-			URL:     horizonURL,
-			HTTP:    http.DefaultClient,
-			AppName: "friendbot",
-		},
+		Secret:            friendbotSecret,
+		Horizon:           hclient,
 		Network:           networkPassphrase,
 		StartingBalance:   startingBalance,
 		SubmitTransaction: internal.AsyncSubmitTransaction,
+		Minions:           minions,
+	}
+}
+
+func createNewAccount(sourceSeed, networkPassphrase string, hclient *horizon.Client) (string, error) {
+	destKeypair, err := keypair.Random()
+	if err != nil {
+		return "", nil
```

we should return a wrapped error instead of nil